mystatus/covid19-reports
# Error handling

Summary of changes
------------------

This PR covers the requirements for the following Asana task:

> [Better input validation and error messages on roster load](https://app.asana.com/0/1188940305683068/1199145504016473)


List of changes
---------------
NOTE: this PR refactors use of ```<AlertDialog>``` and corresponding duplicate code in ```<RosterPage>```

- Added EDIPI validation - 10 digit integer allowing leading zeros
- Added ```value.trim().length``` check in addition to ```value.length``` to ```canSave()```
- Implemented ```<ErrorBoundary>``` component and ```useErrorHandler()``` hook

What code can others reuse?
---------------------------

- ```<ErrorBoundary>``` 
- ```useErrorHandler()```

Should this pull request be tested any more or less than usual?
---------------------------------------------------------------
Due to refactoring ```<AlertDialog>``` use in the ```<RosterPage>```, full regression tests should be performed on error handling.

What are general test steps or cases?
-------------------------------------

1. On Roster page, click 'UPLOAD CSV' button 
2. Select an invalid file
3. Verify the following error dialog:

![Roster-Upload-CSV-Error](https://user-images.githubusercontent.com/73907006/100760691-a5492c80-33b7-11eb-957b-713231045b67.png)

4. Dismiss the dialog and the list is active again, allowing the user to recover and continue.
